### ISSUE TITLE
problem about real depth

### ISSUE THREAD

**Issue by Kaede93** [HUSTSYJ/DA_dahazing#10]
Hello, I am woundering that how can I get the depth images from real hazy images for training the Image translation module? 

Because I noticed that training the S2R-R2S network cycle, we must input the depth images for synthetic hazy images and depth images for real hazy images.

I found that an unlabeled depth dataset is provided, but what if I want to train the network with my own dataset? It seems that there is no guidance for getting this "unlabeled depth", or did I missing something?

Thank you. BTW, 华科nb!

**Comment by Kaede93** [comment]
Oh, I found a depth estimation paper you already quote:

[37] Chuanxia Zheng, Tat-Jen Cham, and Jianfei Cai. T2net:
Synthetic-to-realistic translation for solving single-image
depth estimation tasks. In European Conference on Computer
Vision, 2018. 3

Is that the method for getting the depth images from real hazy images in your paper?

**Comment by Four97** [comment]
have you gotten the real image's depth? 
is ok to test the model with the pictures which dont have depth images?

**Comment by Kaede93** [comment]
> have you gotten the real image's depth?
> is ok to test the model with the pictures which dont have depth images?

Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images. 

I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)

**Comment by Four97** [comment]
> 
> 
> > have you gotten the real image's depth?
> > is ok to test the model with the pictures which dont have depth images?
> 
> Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> 
> I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)

Thanks a lot. I will try it latter.

**Comment by Unitshockresponse** [comment]
Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?

**Comment by rgbd-zml** [comment]
> > have you gotten the real image's depth?
> > is ok to test the model with the pictures which dont have depth images?
> 
> Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> 
> I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)

Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?

**Comment by Kaede93** [comment]
> > > have you gotten the real image's depth?
> > > is ok to test the model with the pictures which dont have depth images?
> > 
> > 
> > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> 
> Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?

Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.

**Comment by libertyllh** [comment]
> > > > have you gotten the real image's depth?
> > > > is ok to test the model with the pictures which dont have depth images?
> > > 
> > > 
> > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > 
> > 
> > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> 
> Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.
In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?

**Comment by libertyllh** [comment]
> > > > have you gotten the real image's depth?
> > > > is ok to test the model with the pictures which dont have depth images?
> > > 
> > > 
> > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > 
> > 
> > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> 
> Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.



> > > > have you gotten the real image's depth?
> > > > is ok to test the model with the pictures which dont have depth images?
> > > 
> > > 
> > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > 
> > 
> > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> 
> Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.

>>In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?

**Comment by libertyllh** [comment]
> > > > have you gotten the real image's depth?
> > > > is ok to test the model with the pictures which dont have depth images?
> > > 
> > > 
> > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > 
> > 
> > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> 
> Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.

In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?

**Comment by Kaede93** [comment]
> > > > > have you gotten the real image's depth?
> > > > > is ok to test the model with the pictures which dont have depth images?
> > > > 
> > > > 
> > > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > > 
> > > 
> > > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> > 
> > 
> > Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.
> 
> In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?

The equation (5) and (6) of this paper mentioned the real image depth map, which is used for training the translation network (for getting cycle consistency loss for S2R and R2S). The open source code also used the real image depth map. You can check this part for getting more information.

In fact, I don't precisely remember which part of this paper mentioned the real image depth map since my notes left behind in my laboratory (I will renew this comment on workday if necessary).

**Comment by liyouxing** [comment]
> > > > > have you gotten the real image's depth?
> > > > > is ok to test the model with the pictures which dont have depth images?
> > > > 
> > > > 
> > > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > > 
> > > 
> > > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> > 
> > 
> > Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.
> 
> In this paper, I did not see the requirements of real image depth map. What is its function? Is it necess

> > > > > > have you gotten the real image's depth?
> > > > > > is ok to test the model with the pictures which dont have depth images?
> > > > > 
> > > > > 
> > > > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > > > 
> > > > 
> > > > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> > > 
> > > 
> > > Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.
> > 
> > 
> > In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?
> 
> The equation (5) and (6) of this paper mentioned the real image depth map, which is used for training the translation network (for getting cycle consistency loss for S2R and R2S). The open source code also used the real image depth map. You can check this part for getting more information.
> 
> In fact, I don't precisely remember which part of this paper mentioned the real image depth map since my notes left behind in my laboratory (I will renew this comment on workday if necessary).


> > > > > have you gotten the real image's depth?
> > > > > is ok to test the model with the pictures which dont have depth images?
> > > > 
> > > > 
> > > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > > 
> > > 
> > > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> > 
> > 
> > Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.
> 
> In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?



> > > > > > have you gotten the real image's depth?
> > > > > > is ok to test the model with the pictures which dont have depth images?
> > > > > 
> > > > > 
> > > > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > > > 
> > > > 
> > > > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> > > 
> > > 
> > > Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.
> > 
> > 
> > In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?
> 
> The equation (5) and (6) of this paper mentioned the real image depth map, which is used for training the translation network (for getting cycle consistency loss for S2R and R2S). The open source code also used the real image depth map. You can check this part for getting more information.
> 
> In fact, I don't precisely remember which part of this paper mentioned the real image depth map since my notes left behind in my laboratory (I will renew this comment on workday if necessary).

i think this method don't need real image depth and there is also no mention of this need in the paper, in code, although have real image depth list in dataloader, i think this list maybe a null list, and it only define in it!

**Comment by liyouxing** [comment]
> > > > > > have you gotten the real image's depth?
> > > > > > is ok to test the model with the pictures which dont have depth images?
> > > > > 
> > > > > 
> > > > > Hi, I sent an email to one of the authors Shao and he replied that they used the method in "MegaDepth: Learning Single-View Depth Prediction from Internet Photos"(2018 CVPR) for obtaining the depth images.
> > > > > I haven't try yet because I'm struggled with other issues, please contact me if you done the jobs :)
> > > > 
> > > > 
> > > > Does the cycleGAN network need a depth map of a real picture or a depth map of a synthesized picture?
> > > 
> > > 
> > > Are we talking about the vanilla CycleGAN? The depth map was unnecessary for vanilla CycleGAN. In fact, it was doing the image to image domain transfer without any physical model, you can just transfer the hazy images to the clean ones.
> > 
> > 
> > In this paper, I did not see the requirements of real image depth map. What is its function? Is it necessary?
> 
> The equation (5) and (6) of this paper mentioned the real image depth map, which is used for training the translation network (for getting cycle consistency loss for S2R and R2S). The open source code also used the real image depth map. You can check this part for getting more information.
> 
> In fact, I don't precisely remember which part of this paper mentioned the real image depth map since my notes left behind in my laboratory (I will renew this comment on workday if necessary).

i think this method don't need real image depth and there is also no mention of this need in the paper, in code, although have real image depth list in dataloader, i think this list maybe a null list, and it only define in it!

**Comment by wushunshun** [comment]
Hello, have you gotten the real image's depth?
is ok to test the model with the pictures which dont have depth images?

**Comment by Four97** [comment]
yeah, you can test images without depth information, because in Dehazed Network, we dont need it.



At 2021-01-03 15:02:12, "wushunshun" <notifications@github.com> wrote:

Hello, have you gotten the real image's depth?
is ok to test the model with the pictures which dont have depth images?
